Website/Providers/DataProviders/SqlDataProvider/09.04.00.01.SqlDataProvider

```diff
@@ -0,0 +1,168 @@
+	@PortalID	INT,
+	@Rolename	NVARCHAR(50)
+AS
+	DECLARE @UserPortalId INT
+	DECLARE @PortalGroupId INT
+	SELECT @PortalGroupId = PortalGroupId FROM {databaseOwner}[{objectQualifier}Portals] WHERE PortalID = @PortalID
+	IF EXISTS(SELECT PortalGroupID FROM {databaseOwner}[{objectQualifier}PortalGroups] WHERE PortalGroupID = @PortalGroupId)
+	BEGIN
+		SELECT @UserPortalId = MasterPortalID FROM {databaseOwner}[{objectQualifier}PortalGroups] WHERE PortalGroupID = @PortalGroupId
+	END
+	ELSE
+	BEGIN
+		SELECT @UserPortalId = @PortalID
+	END
+	SELECT     
+		U.*, 
+		UP.PortalId, 
+		UP.Authorised, 
+    UP.HasAgreedToTerms,
+    UP.HasAgreedToTermsOn,
+		UP.IsDeleted,
+		UP.RefreshRoles,
+		UP.VanityUrl
+	FROM {databaseOwner}{objectQualifier}UserPortals AS UP 
+			RIGHT OUTER JOIN {databaseOwner}{objectQualifier}UserRoles  UR 
+			INNER JOIN {databaseOwner}{objectQualifier}Roles R ON UR.RoleID = R.RoleID 
+			RIGHT OUTER JOIN {databaseOwner}{objectQualifier}Users AS U ON UR.UserID = U.UserID 
+		ON UP.UserId = U.UserID	
+	WHERE ( UP.PortalId = @UserPortalId OR @UserPortalId IS Null )
+		AND (UP.IsDeleted = 0 OR UP.IsDeleted Is NULL)
+		AND (R.RoleName = @Rolename)
+		AND (R.PortalId = @PortalID OR @PortalID IS Null )
+	ORDER BY U.FirstName + ' ' + U.LastName
+
+GO
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}UserAgreedToTerms]') AND type in (N'P', N'PC'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}UserAgreedToTerms]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}UserAgreedToTerms]
+	@PortalId	INT,
+	@UserId		INT
+AS
+	UPDATE {databaseOwner}{objectQualifier}UserPortals
+	SET HasAgreedToTerms = 1, HasAgreedToTermsOn = GETDATE()
+	WHERE PortalId = @PortalId AND UserId = @UserId
+GO
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}ResetTermsAgreement]') AND type in (N'P', N'PC'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}ResetTermsAgreement]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}ResetTermsAgreement]
+	@PortalId	INT
+AS
+	UPDATE {databaseOwner}{objectQualifier}UserPortals
+	SET HasAgreedToTerms = 0
+	WHERE PortalId = @PortalId
+GO
+
+/*      Add logging for consent             */
+/********************************************/
+
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogTypes] ([LogTypeKey], [LogTypeFriendlyName], [LogTypeDescription], [LogTypeOwner], [LogTypeCSSClass]) 
+VALUES (N'DATA_CONSENT_SUCCESS', N'User Consented', N'', N'DotNetNuke.Logging.EventLogType', N'OperationSuccess')
+GO
+
+INSERT INTO {databaseOwner}[{objectQualifier}EventLogTypes] ([LogTypeKey], [LogTypeFriendlyName], [LogTypeDescription], [LogTypeOwner], [LogTypeCSSClass]) 
+VALUES (N'DATA_CONSENT_FAILURE', N'User Failed to Consent', N'', N'DotNetNuke.Logging.EventLogType', N'AdminAlert')
+GO
+
+IF NOT EXISTS (SELECT * FROM {databaseOwner}{objectQualifier}EventLogConfig WHERE LogTypeKey = 'DATA_CONSENT_SUCCESS')
+BEGIN  
```

Comment:
What is the benefit of just keeping the past 10 records?
IMO success records should only be kept, if there is a concrete need - otherwise logging should turned off.

Reply:
@sleupold This is a good idea.  @donker Possibly disabled by default?  Thoughts.  Users could turn this on, but the data is in fact recorded elsewhere

Reply:
OK. Done. It's gone.